plugins/twostepauth.php

```diff
@@ -697,7 +697,7 @@ function twostepauth_login()
     global $user, $db, $mybb, $footer, $header, $navigation, $headerinclude, $themes, $templates, $usercpnav, $lang, $session, $logins, $twostepauth_loginattempts;
 
     if($user == null) return;
-    $user_info = $db->fetch_array($db->simple_select("users", "*", "uid = ".$user["uid"]));
+    $user_info = $db->fetch_array($db->simple_select("users", "*", "username = '".$user["username"]."'"));
```

Comment:
Why don't you escape this? Also note that some DB configurations may require you to search on lower case letters.